volumes.cpp

```diff
@@ -1059,23 +1070,31 @@ bool sqIsEllipsoid(const std::vector< double >& args) {
 // the VolumeCache to use if none is provided to makeSurface() calls.
 static VolumeCache default_volume_cache;
 
-SurfaceVolume& makeSurface( const SurfaceCard* card, VolumeCache* v){
-  
+// Facet number is 0 if not provided
+SurfaceVolume& makeSurface( const SurfaceCard* card, VolumeCache* v, int facet){
   VolumeCache& cache = default_volume_cache;
+  SurfaceVolume* surface;
   if( v != NULL ){
     cache = *v;
   }
 
   
-  if( cache.contains( card ) ){
+  if( cache.contains( card ) && facet == 0 ){
```

Comment:
I think we need to think through how this cache works.  It looks like you now cache facet surfaces (see line 1324 or so) so it seems like you should use them, but the cache key needs to be rethought.

It might be exposed if you had a test (maybe you do?) that uses the full macrobody and also uses a facet from that macrobody more than once, as well.